File: go/concurrent_tasks.go
package main

import (
	"fmt"
	"sync"
	"time"
)

func task(number int, wg *sync.WaitGroup) {
	defer wg.Done()

	fmt.Printf("Task %d started\n", number)

	time.Sleep(1 * time.Second)

	fmt.Printf("Task %d finished\n", number)
}

func main() {
	startTime := time.Now()

	var wg sync.WaitGroup

	for i := 1; i <= 5; i++ {
		wg.Add(1)
		go task(i, &wg)
	}

	wg.Wait()

	elapsed := time.Since(startTime)

	fmt.Printf("Elapsed time: %.2f seconds\n", elapsed.Seconds())
}

File: python/concurrent_tasks.py
import asyncio
import time


async def task(number):
    print(f"Task {number} started")
    await asyncio.sleep(1)
    print(f"Task {number} finished")


async def main():
    start_time = time.time()

    await asyncio.gather(
        task(1),
        task(2),
        task(3),
        task(4),
        task(5)
    )

    elapsed = time.time() - start_time
    print("Elapsed time:", round(elapsed, 2), "seconds")


asyncio.run(main())
